### PR TITLE
[BUG][Connector-V2][Mongo-cdc] Incremental data kind error in snapshot phase

### LINKED ISSUE #5169: [Bug] [MongoDB-CDC] Incremental data kind error in snapshot phase
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Incremental data kind error in snapshot phase

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
none
```

### Running Command

```shell
none
```

### Error Exception

```log
none
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

cdc-base logic
image
image

MongoDB cdc logic
image

trigger error location
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
…t phase

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

close #5169

### CODE REVIEW

**Comment by Carl-Zhou-CN** [comment]
@hailin0 @MonsterChenzhuo Please help review

**Comment by MonsterChenzhuo** [comment]
I think it would be better for you to associate an issue and explain the problem.

**Comment by MonsterChenzhuo** [comment]
> @hailin0 @MonsterChenzhuo Please help review

I think it would be better for you to associate an issue and explain the problem.

**Comment by Carl-Zhou-CN** [comment]
> > @hailin0 @MonsterChenzhuo Please help review
> 
> I think it would be better for you to associate an issue and explain the problem.

@MonsterChenzhuo ,Sorry, the reason for this pr is that all data in the snapshot phase should have a rowkind of read

**Comment by ic4y** [comment]
@MonsterChenzhuo PTAL

**Comment by hailin0** [comment]
good pr

**Comment by hailin0** [comment]
waiting for local test run
@Carl-Zhou-CN

**Comment by EricJoy2048** [comment]
@MonsterChenzhuo  PTAL

**Comment by MonsterChenzhuo** [comment]
@Carl-Zhou-CN It's like this: MySQL's binary log contains "R" events, and maintaining consistency between full and incremental backups should be approached in a certain way. However, MongoDB's oplog (operational log) does not have "R" events. Therefore, employing the same methodology is incorrect. This discrepancy is connected to the event types within MongoDB's oplog (or change streams).

**Comment by MonsterChenzhuo** [comment]
@Carl-Zhou-CN  Let's go ahead and merge the 'dev' branch first. Due to the issue mentioned in https://github.com/apache/seatunnel/pull/5195, the code is unable to trigger the 'rewriteOutputBuffer' function at all. Therefore, I'd like to debug and see if there's any improvement in the situation.

**Comment by MonsterChenzhuo** [comment]
企业微信截图_1da302e4-f2f0-4074-b1c9-02494b2847ef
企业微信截图_53a06ef3-d9af-42bb-8860-fbe5868e2b73
企业微信截图_ae6d3d52-0aa3-4930-9ed6-10989fd44355

**Comment by MonsterChenzhuo** [comment]
"I don't understand the relevance between your statement 'all data in the snapshot phase should have a rowkind of read' and the changes you made. Could you please clarify?"

**Comment by MonsterChenzhuo** [comment]
> > > @hailin0 @MonsterChenzhuo Please help review
> > 
> > 
> > I think it would be better for you to associate an issue and explain the problem.
> 
> @MonsterChenzhuo ,Sorry, the reason for this pr is that all data in the snapshot phase should have a rowkind of read

I don't understand the relevance between your statement 'all data in the snapshot phase should have a rowkind of read' and the changes you made. 
you can provide some screenshots and relevant instructions.

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mongodb/source/fetch/MongodbFetchTaskContext.java [hunk 1]
```diff
@@ -172,9 +178,24 @@ public void rewriteOutputBuffer(
...
                 case INSERT:
+                    outputBuffer.put(key, changeRecord);
+                    break;
                 case UPDATE:
                 case REPLACE:
-                    outputBuffer.put(key, changeRecord);
+                    Schema valueSchema = changeRecord.valueSchema();
+                    BsonDocument fullDocument =
+                            extractBsonDocument(value, valueSchema, FULL_DOCUMENT);
+                    fullDocument.put(OPERATION_TYPE, new BsonString(OPERATION_TYPE_INSERT));
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mongodb/source/fetch/MongodbFetchTaskContext.java]: Reset new operation type here

@MonsterChenzhuo

**Comment by Carl-Zhou-CN** [comment]
insert record
{Struct@12258} "Struct{_id={"_id": {"$oid": "100000000000000000000103"}}}" -> {SourceRecord@12259} "SourceRecord{sourcePartition={ns=mongodb://localhost:27017/inventory.products}, sourceOffset={_id={"_id": {"$oid": "100000000000000000000103"}}, copy=true}} ConnectRecord{topic='inventory.products', kafkaPartition=null, key=Struct{_id={"_id": {"$oid": "100000000000000000000103"}}}, keySchema=Schema{keySchema:STRUCT}, value=Struct{_id={"_id": {"$oid": "100000000000000000000103"}},operationType=insert,fullDocument={"_id": {"$oid": "100000000000000000000103"}, "name": "12-pack drill bits", "description": "12-pack of drill bits with sizes ranging from #40 to #3", "weight": "8"},source=Struct{ts_ms=0,table=products,db=inventory,snapshot=true},ts_ms=1691671180997,ns=Struct{db=inventory,coll=products},documentKey={"_id": {"$oid": "100000000000000000000103"}}}, valueSchema=Schema{ChangeStream:STRUCT}, timestamp=null, headers=ConnectHeaders(headers=)}"

update record
SourceRecord{sourcePartition={ns=mongodb://localhost:27017/inventory.products}, sourceOffset={_id={"_data": "8264D4DA89000000032B022C0100296E5A10043A11F44315724484B2AE361B9FAEEB5846645F696400641000000000000000000001030004"}, copy=false}} ConnectRecord{topic='inventory.products', kafkaPartition=null, key=Struct{_id={"_id": {"$oid": "100000000000000000000103"}}}, keySchema=Schema{keySchema:STRUCT}, value=Struct{_id={"_id": {"$oid": "100000000000000000000103"}},operationType=insert,fullDocument={"_id": {"$oid": "100000000000000000000103"}, "name": "12-pack drill bits", "description": "Set of 12 professional-grade drill bits", "weight": "8"},source=Struct{ts_ms=0,snapshot=true},ts_ms=1691671183383,ns=Struct{db=inventory,coll=products},documentKey={"_id": {"$oid": "100000000000000000000103"}}}, valueSchema=Schema{ChangeStream:STRUCT}, timestamp=null, headers=ConnectHeaders(headers=)}


SeaTunnelRow#tableId= SeaTunnelRow#kind=INSERT : 100000000000000000000103, 12-pack drill bits, Set of 12 professional-grade drill bits, 8

**Comment by Carl-Zhou-CN** [comment]
企业微信截图_1691671305298
企业微信截图_16916713546247

**Comment by ic4y** [comment]
@hailin0 @MonsterChenzhuo @Carl-Zhou-CN What's the status of this PR now?

**Review by hailin0 — APPROVED** [review]

**Comment by MonsterChenzhuo** [comment]
LGTM

**Review by ic4y — APPROVED** [review]

**Pull request merged**